airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/initialsync/MssqlInitialLoadHandler.java

```diff
@@ -0,0 +1,161 @@
+      final JdbcDatabase database,
+      final MssqlSourceOperations sourceOperations,
+      final String quoteString,
+      final MssqlInitialLoadStateManager initialLoadStateManager,
+      final Function<AirbyteStreamNameNamespacePair, JsonNode> streamStateForIncrementalRunSupplier) {
+    this.config = config;
+    this.database = database;
+    this.sourceOperations = sourceOperations;
+    this.quoteString = quoteString;
+    this.initialLoadStateManager = initialLoadStateManager;
+    this.streamStateForIncrementalRunSupplier = streamStateForIncrementalRunSupplier;
+  }
+  public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(
+      final ConfiguredAirbyteCatalog catalog,
+      final Map<String, TableInfo<CommonField<JDBCType>>> tableNameToTable,
+      final Instant emittedAt) {
+    final List<AutoCloseableIterator<AirbyteMessage>> iteratorList = new ArrayList<>();
+    for (final ConfiguredAirbyteStream airbyteStream : catalog.getStreams()) {
+      final AirbyteStream stream = airbyteStream.getStream();
+      final String streamName = stream.getName();
+      final String namespace = stream.getNamespace();
+      // TODO: need to select column according to indexing status of table. may not be primary key
+      final List<String> primaryKeys = stream.getSourceDefinedPrimaryKey().stream().flatMap(pk -> Stream.of(pk.get(0))).toList();
+      final AirbyteStreamNameNamespacePair pair = new AirbyteStreamNameNamespacePair(streamName, namespace);
+      final String fullyQualifiedTableName = DbSourceDiscoverUtil.getFullyQualifiedTableName(namespace, streamName);
+      if (!tableNameToTable.containsKey(fullyQualifiedTableName)) {
+        LOGGER.info("Skipping stream {} because it is not in the source", fullyQualifiedTableName);
+        continue;
+      }
+      if (airbyteStream.getSyncMode().equals(SyncMode.INCREMENTAL)) {
+        // Grab the selected fields to sync
+        final TableInfo<CommonField<JDBCType>> table = tableNameToTable.get(fullyQualifiedTableName);
+        final List<String> selectedDatabaseFields = table.getFields()
+            .stream()
+            .map(CommonField::getName)
+            .filter(CatalogHelpers.getTopLevelFieldNames(airbyteStream)::contains)
+            .toList();
+        primaryKeys.forEach(pk -> {
+          if (!selectedDatabaseFields.contains(pk)) {
+            selectedDatabaseFields.add(0, pk);
+          }
+        });
+
+        final AutoCloseableIterator<JsonNode> queryStream =
+            new MssqlInitialLoadRecordIterator(database, sourceOperations, quoteString, initialLoadStateManager, selectedDatabaseFields, pair, calculateChunkSize(), isCompositePrimaryKey(airbyteStream));
+        final AutoCloseableIterator<AirbyteMessage> recordIterator =
+            getRecordIterator(queryStream, streamName, namespace, emittedAt.toEpochMilli());
+        final AutoCloseableIterator<AirbyteMessage> recordAndMessageIterator = augmentWithState(recordIterator, pair);
```

Comment:
nit: can we use `var` more? the LHS types in this line and those above is not ambiguous